homeassistant/components/arcticspa/hottub.py

```diff
@@ -0,0 +1,31 @@
+"""hottub contains the Device class that represents an Arctic Spa Hot Tub."""
+from hashlib import sha256
+from http import HTTPStatus
+from urllib.error import HTTPError
+
+from pyarcticspas import Spa
+
+
+class Device:
+    """Device is an Arctic Spa Hot Tub identified by its API key."""
+
+    def __init__(self, api_key: str) -> None:
+        """Initialize Spa Entry."""
+        self.api = Spa(api_key)
+        self.id = sha256(bytes(api_key, "utf8")).hexdigest()
```

Comment:
Oh I now see the device id is this. Can we get an id from the API? if the API key ever expires all entities get doubled and automations will break.

Reply:
Unfortunately, the API is fairly simple and does not handle sessions at all. You always have to provide the fixed API key for every request.

But a good point is made, I moved the whole logic into the underlying library. When they update the API to have IDs, it will be less intrusive for the Home Assistant codebase.